substrate/client/transaction-pool/src/graph/validated_pool.rs

```diff
@@ -711,6 +741,11 @@ impl<B: ChainApi> ValidatedPool<B> {
 		log::trace!(target: LOG_TARGET, "Removed invalid transactions: {:?}/{:?}", hashes.len(), invalid.len());
 		log_xt_trace!(target: LOG_TARGET, invalid.iter().map(|t| t.hash), "Removed invalid transaction");
 
+		let mut event_dispatcher = self.event_dispatcher.write();
+		for tx in &invalid {
+			event_dispatcher.invalid(&tx.hash);
```

Comment:
If I see correctly we report these invelid events twice now, above we already have the listener action that does this.

Reply:
good catch :see_no_evil: , thanks!

Reply:
fixed in 0b5d6b75c52